app/src/main/java/com/example/tictactoe/MainActivity.java

```diff
@@ -2,102 +2,153 @@
+//        Button btn03 = findViewById(R.id.btn_3);
+//        Button btn04 = findViewById(R.id.btn_4);
+//        Button btn05 = findViewById(R.id.btn_5);
+//        Button btn06 = findViewById(R.id.btn_6);
+//        Button btn07 = findViewById(R.id.btn_7);
+//        Button btn08 = findViewById(R.id.btn_8);
+//
+//        btns.add(btn00);
+//        btns.add(btn01);
+//        btns.add(btn02);
+//        btns.add(btn03);
+//        btns.add(btn04);
+//        btns.add(btn05);
+//        btns.add(btn06);
+//        btns.add(btn07);
+//        btns.add(btn08);
+//    }//end onCreate method
+//
+
+
+        //theoretically we should be doing the same thing for every btn in the table
+        //To access the array list of btns, we simply check which button was pressed,
+        //then do whatever we want to do with each of the buttons
+        //this saves time copying and individualizing what we want done with each button
+//    public void tableBtnCLicked(int btn)
+//    {
+//        //for example
+//        for (int i = 0; i < btns.size(); i++)
+//        {
+//            if(i == btn)
+//            {
+//                //not the actual algorithm, just a quick example
+//                btns.get(i).setText("X");
+//            }
+//        }
+//    }
+
+    }//end MainActivity
+
+    @Override
+    public void onClick(View view) {
+        if (!((Button) view).getText().toString().equals("")) {
+            return;
+        }
+        String buttonID = view.getResources().getResourceEntryName(view.getId());
+        int gameStatePointer = Integer.parseInt(buttonID.substring(buttonID.length() - 1, buttonID.length()));
+
+        if (activePlayer){
+            ((Button) view).setText("X");
+        ((Button) view).setTextColor(Color.parseColor("#465362"));
+        gameState[gameStatePointer] = 0;
+    } else {
+            ((Button)view).setText("O");
+            ((Button)view).setTextColor(Color.parseColor("#82A3A1"));
+            gameState[gameStatePointer] = 1;
+        }
+        rountCount++;
+
+        if(checkWinner())
+        {
+            if(activePlayer)
+            {
+                playerOneScoreCount++;
+                updatePlayerScore();
+            } else {
+                playerTwoScoreCount++;
+                updatePlayerScore();
+            }
+        } else if (rountCount==9)
+        {
+            //put play again in here
+        } else {
+            activePlayer = !activePlayer;
+        }
+
+    }
+    public boolean checkWinner()
     {
-        //for example
-        for (int i = 0; i < btns.size(); i++)
+        boolean winnerResult = false;
+        for(int [] winningPosition : winningPositions)
         {
-            if(i == btn)
+            if(gameState[winningPosition[0]] == gameState[winningPosition[1]] &&
+                    gameState[winningPosition[1]] == gameState[winningPosition[2]] &&
+                    gameState[winningPosition[0]] != 2)
             {
```

Comment:
Please comment throughout your code so that we can all understand your thought process. I can follow most of it, but we each have a specific style of coding so it helps if you add comments (as you saw I did in the previous commit). I really like your algorithm here to check the winning button positions - though to be honest I'm not very good with 2D arrays and could probably do with a couple comments.